edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProcessor.cs

```diff
@@ -140,17 +140,17 @@ public void Dispose()
                 .Select(b => new ArraySegment<byte>(b.ToArray()));
 
         static Source<ArraySegment<byte>, AkkaNet.NotUsed> NonGroupingGzipMapper(Source<ArraySegment<byte>, AkkaNet.NotUsed> s) =>
-            s.Select(m => new ArraySegment<byte>(Compression.CompressToGzip(m.Array)));
+            s.Select(m => new ArraySegment<byte>(Compression.CompressToGzip(m)));
 
         static Source<ArraySegment<byte>, AkkaNet.NotUsed> GroupingGzipMapper(Source<ArraySegment<byte>, AkkaNet.NotUsed> s, LogsOutputGroupingConfig outputGroupingConfig) =>
             s.GroupedWithin(outputGroupingConfig.MaxFrames, outputGroupingConfig.MaxDuration)
                 .Select(
                     b =>
                     {
-                        var combinedArray = b.Select(a => a.Array).ToList().Combine();
+                        var combinedArray = b.SelectMany(a => a).ToArray();
```

Comment:
@varunpuranik please verify that this is the proper behavior. This makes the function grab only the part of the array represented by the array segment instead of the underlying array. We discussed, and you said this seems like the proper behavior, but I want to triple check.